projects/vfat/utils.js

```diff
@@ -0,0 +1,72 @@
+const { Balances } = require('@defillama/sdk');
+const { sumTokens2 } = require('../helper/unwrapLPs');
+
+async function sumLPBalances(api, gauges, sickles, lpTokens) {
+  api.log(api.chain, sickles.length, gauges.length,)
+  lpTokens.forEach((lpToken, index) => lpTokens[index] = lpToken.toLowerCase())
+  let minLPValue = 4e3
+  if (lpTokens.length > 200) minLPValue = 15e3
+  if (lpTokens.length > 400) minLPValue = 25e3
+
+  const filteredLPSet = new Set(await filteredLPTokens({ api, lpTokens, minLPValue, }))
+
+  const gaugeTokenMapping = {};
+  const tokens = []
+  lpTokens.forEach((lpToken, index) => {
+    if (!filteredLPSet.has(lpToken)) return;
+
+    const gauge = gauges[index].toLowerCase()
+    tokens.push(gauge)
+    gaugeTokenMapping[gauge] = lpToken
+  });
+
+  const transformAddress = (token) => {
+    token = token.toLowerCase()
+    return `${api.chain}:${gaugeTokenMapping[token] ?? token}`
+  }
+  await sumTokens2({ api, transformAddress, tokens, owners: sickles })
+}
+
+// we are going to filter out tokens that we dont have price in the server and LP tokens with less than 1k value in it
+async function filteredLPTokens({ api, lpTokens, minLPValue = 10e3 }) {
```

Comment:
other issue was, we were checking against every gauge for every sickle, since we dont know what all gauges a sickle has, what I did here was, filter out low value LP tokens to cut down on the # calls